src/distributors/local-installer.ts

```diff
@@ -0,0 +1,63 @@
+import * as tc from '@actions/tool-cache';
+
+import fs from 'fs';
+import path from 'path';
+import semver from 'semver';
+import { macOSJavaContentDir, extractJdkFile } from '../util';
+
+import { JavaBase } from './base-installer';
+import { JavaInstallerOptions, JavaDownloadRelease, JavaInstallerResults } from './base-models';
+
+export class LocalDistributor extends JavaBase {
+  constructor(installerOptions: JavaInstallerOptions, private jdkFile?: string) {
+    super('LocalJDKFile', installerOptions);
+  }
+
+  public async setupJava(): Promise<JavaInstallerResults> {
+    let foundJava = this.findInToolcache();
+
+    if (!foundJava) {
+      if (!this.jdkFile) {
+        throw new Error("'jdkFile' is not specified");
+      }
+      const jdkFilePath = path.resolve(this.jdkFile);
+      const stats = fs.statSync(jdkFilePath);
+
+      if (!stats.isFile()) {
+        throw new Error(`JDK file is not found in path '${jdkFilePath}'`);
+      }
+
+      const extractedJavaPath = await extractJdkFile(jdkFilePath);
```

Comment:
Add core.info `Extracting Java from '${jdkFilePath}'`

Reply:
Done. Thank you.